### PR TITLE
nightly pipeline build using PTCA image.

### PR DESCRIPTION
**Pull request by AdamLouly**
**Description**: 
Nightly pipeline build.

**Motivation and Context**
- This pipeline will **NOT** get triggered with PRs and will only run when its scheduled.
- The purpose is to catch the failures and errors on the nightly ort version instead of waiting for the stable version, this way we can solve these errors upfront while we have time.
- This pipeline is projected to run on daily basis, but for now It will only run on a weekly basis, because it pulls the PTCA nightly image and this image gets generated every Friday,
The PTCA team is working on making it get generated on daily basis, until then the pipeline will be scheduled to run once a week (after the PTCA nightly image generation).
- The previous PR (https://github.com/microsoft/onnxruntime/pull/12277) was closed and replaced with this one.
- The pipeline was tested today 15th August 2022 after the PTCA image was generated on Friday, and it ran successfully ( [Pipeline Logs ](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=722096&view=logs&j=a242bf24-7d40-50e3-91cc-3075088d1dc9&t=a242bf24-7d40-50e3-91cc-3075088d1dc9))

### CODE REVIEW

**Review by snnn — APPROVED** [review]

File: tools/ci_build/github/azure-pipelines/orttraining-linux-nightly-gpu-on-device-training.yml
Comment (askhade) [inline tools/ci_build/github/azure-pipelines/orttraining-linux-nightly-gpu-on-device-training.yml:23]: On-Device ?

**Comment by AdamLouly** [comment]
Removed torchtext installing as the new PTCA will include that, it will get generated tomorrow, PR can be merged once the pipelines are all ran successfully.

File: tools/ci_build/github/azure-pipelines/orttraining-linux-nightly-ortmodule-test-pipeline.yml
Comment (baijumeswani) [inline tools/ci_build/github/azure-pipelines/orttraining-linux-nightly-ortmodule-test-pipeline.yml:21]: Considering that we are going to stop publishing cuda 11.3 package, shall we request PTCA team to give us a 11.6 package?

Comment (askhade) [reply tools/ci_build/github/azure-pipelines/orttraining-linux-nightly-ortmodule-test-pipeline.yml:21]: Sure. But we don't need to block this PR. Let's merge this.

**Review by askhade — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Comment by AdamLouly** [comment]
Thanks for the Review.

**Pull request merged**
